Source/WebCore/loader/cache/CachedFont.cpp

```diff
@@ -64,10 +68,47 @@ void CachedFont::didAddClient(CachedResourceClient& client)
         downcast<CachedFontClient>(client).fontLoaded(*this);
 }
 
+static AtomString hashForFontData(const uint8_t* data, size_t size)
+{
+    auto cryptoDigest = PAL::CryptoDigest::create(PAL::CryptoDigest::Algorithm::SHA_256);
+    cryptoDigest->addBytes(data, size);
+    auto digest = cryptoDigest->computeHash();
+    return AtomString(makeString(base64Encoded(digest.data(), digest.size())));
+}
+
+static bool isAllowedFont(const AtomString& fontSha256)
+{
+    return allowedHashes().contains(fontSha256);
+}
+
+bool CachedFont::shouldAllowCustomFont(const Ref<SharedBuffer>& data)
+{
+    if (!m_loader || !m_loader->frame())
+        return false;
+
+    switch (m_loader->frame()->settings().downloadableBinaryFontAllowedTypes()) {
+    case DownloadableBinaryFontAllowedTypes::Any:
+        return true;
+    case DownloadableBinaryFontAllowedTypes::None:
+        return false;
+    case DownloadableBinaryFontAllowedTypes::Restricted: {
+        auto sha = hashForFontData(data->data(), data->size());
```

Comment:
We may want to add a `dataAsSpan()` function to SharedBuffer that returns a `Span<const uint8_t>`, and use it here.